platform-sdk/consensus-otter-tests/src/testFixtures/java/org/hiero/otter/fixtures/container/ContainerNode.java

```diff
@@ -0,0 +1,286 @@
+         */
+        public ContainerAsyncNodeActions(@NonNull final Duration timeout) {
+            this.timeout = timeout;
+        }
+
+        /**
+         * {@inheritDoc}
+         */
+        @Override
+        public void start() {
+            throwIfIn(LifeCycle.RUNNING, "Node has already been started.");
+            throwIfIn(LifeCycle.DESTROYED, "Node has already been destroyed.");
+
+            log.info("Starting node {}...", selfId);
+
+            final StartRequest startRequest = StartRequest.newBuilder()
+                    .setSelfId(ProtobufConverter.toGoogle(selfId))
+                    .setRoster(ProtobufConverter.toGoogle(roster))
+                    .setKeysAndCerts(KeysAndCertsConverter.toProto(keysAndCerts))
+                    .putAllOverriddenProperties(nodeConfiguration.overriddenProperties())
+                    .build();
+
+            final TestControlStub stub = TestControlGrpc.newStub(channel);
+            stub.start(startRequest, new StreamObserver<>() {
+                @Override
+                public void onNext(final EventMessage value) {
+                    if (value.hasPlatformStatusChange()) {
+                        final PlatformStatusChange change = value.getPlatformStatusChange();
+                        final String statusName = change.getNewStatus();
+                        try {
+                            final PlatformStatus newStatus = PlatformStatus.valueOf(statusName);
+                            platformStatus = newStatus;
+                            resultsCollector.addPlatformStatus(newStatus);
+                        } catch (final IllegalArgumentException e) {
+                            log.warn("Received unknown platform status: {}", statusName);
+                        }
+                    } else if (value.hasLogEntry()) {
+                        receivedLogs.add(ProtobufConverter.fromGoogle(value.getLogEntry()));
+                    } else if (value.hasConsensusRounds()) {
+                        resultsCollector.addConsensusRounds(ProtobufConverter.fromGoogle(value.getConsensusRounds()));
+                    } else {
+                        // Other message types can be handled later
+                        log.info(value);
+                    }
+                }
+
+                @Override
+                public void onError(@NonNull final Throwable t) {
+                    log.error("gRPC error from node {}", selfId, t);
+                }
+
+                @Override
+                public void onCompleted() {
+                    // nothing
+                }
+            });
+
+            lifeCycle = RUNNING;
+        }
+
+        /**
+         * {@inheritDoc}
+         */
+        @Override
+        public void killImmediately() {
+            log.info("Killing node {} immediately...", selfId);
+            if (channel != null) {
+                channel.shutdownNow();
+            }
+            container.stop();
```

Comment:
This is probably not correct. We have to be able to restart the node, which means stopping the container might be too much. We will see once we implement the [freeze functionality](https://github.com/hiero-ledger/hiero-consensus-node/issues/19717).